{
  "uid" : "b2fd2386b1da2c23",
  "name" : "TC 02.02.05 - Check Contact Us link in footer leads to the correct page from each page specified in DATA_1",
  "fullName" : "tests.test_footer.TestFooter#test_tc_02_02_05_check_contact_us_link_functionality",
  "historyId" : "94f0e1cf98dd00a5517d7dafd8a82281",
  "time" : {
    "start" : 1707614079442,
    "stop" : 1707614085427,
    "duration" : 5985
  },
  "description" : "Check that Contact Us link in footer is correct",
  "descriptionHtml" : "<p>Check that Contact Us link in footer is correct</p>\n",
  "status" : "broken",
  "statusMessage" : "selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x56120ce80d93 <unknown>\n#1 0x56120cb64337 <unknown>\n#2 0x56120cbafd60 <unknown>\n#3 0x56120cbafe51 <unknown>\n#4 0x56120cbf3bc4 <unknown>\n#5 0x56120cbd246d <unknown>\n#6 0x56120cbf112d <unknown>\n#7 0x56120cbd21e3 <unknown>\n#8 0x56120cba2135 <unknown>\n#9 0x56120cba313e <unknown>\n#10 0x56120ce44e4b <unknown>\n#11 0x56120ce48dfa <unknown>\n#12 0x56120ce316d5 <unknown>\n#13 0x56120ce49a6f <unknown>\n#14 0x56120ce1569f <unknown>\n#15 0x56120ce6e098 <unknown>\n#16 0x56120ce6e262 <unknown>\n#17 0x56120ce7ff34 <unknown>\n#18 0x7fcb15c94ac3 <unknown>",
  "statusTrace" : "self = <test_footer.TestFooter object at 0x7f4e6057e350>\ndriver = <selenium.webdriver.chrome.webdriver.WebDriver (session=\"d5cfcff0714348071d8c97ebbeacd3f8\")>\nURL = 'https://magento.softwaretestingboard.com/what-is-new.html'\n\n    @allure.title(\"TC 02.02.05 - Check Contact Us link in footer leads to the correct page \"\n                  \"from each page specified in DATA_1\")\n    @pytest.mark.parametrize('URL', DATA_1)\n    def test_tc_02_02_05_check_contact_us_link_functionality(self, driver, URL):\n        \"\"\"Check that Contact Us link in footer is correct\"\"\"\n        page = FooterPage(driver, url=URL)\n        page.open()\n>       page.check_contact_us_link_functionality()\n\ntests/test_footer.py:557: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \npages/footer_page.py:326: in check_contact_us_link_functionality\n    contact_us_link = self.element_is_visible(self.footer_locators.CONTACT_US_LINK)\npages/base_page.py:29: in element_is_visible\n    self.go_to_element(self.element_is_present(locator))\npages/base_page.py:50: in element_is_present\n    return wait(self.driver, timeout).until(EC.presence_of_element_located(locator))\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"d5cfcff0714348071d8c97ebbeacd3f8\")>\nmethod = <function presence_of_element_located.<locals>._predicate at 0x7f4e60926700>\nmessage = ''\n\n    def until(self, method, message: str = \"\"):\n        \"\"\"Calls the method provided with the driver as an argument until the \\\n        return value does not evaluate to ``False``.\n    \n        :param method: callable(WebDriver)\n        :param message: optional message for :exc:`TimeoutException`\n        :returns: the result of the last call to `method`\n        :raises: :exc:`selenium.common.exceptions.TimeoutException` if timeout occurs\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.monotonic() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, \"screen\", None)\n                stacktrace = getattr(exc, \"stacktrace\", None)\n            time.sleep(self._poll)\n            if time.monotonic() > end_time:\n                break\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message: \nE       Stacktrace:\nE       #0 0x56120ce80d93 <unknown>\nE       #1 0x56120cb64337 <unknown>\nE       #2 0x56120cbafd60 <unknown>\nE       #3 0x56120cbafe51 <unknown>\nE       #4 0x56120cbf3bc4 <unknown>\nE       #5 0x56120cbd246d <unknown>\nE       #6 0x56120cbf112d <unknown>\nE       #7 0x56120cbd21e3 <unknown>\nE       #8 0x56120cba2135 <unknown>\nE       #9 0x56120cba313e <unknown>\nE       #10 0x56120ce44e4b <unknown>\nE       #11 0x56120ce48dfa <unknown>\nE       #12 0x56120ce316d5 <unknown>\nE       #13 0x56120ce49a6f <unknown>\nE       #14 0x56120ce1569f <unknown>\nE       #15 0x56120ce6e098 <unknown>\nE       #16 0x56120ce6e262 <unknown>\nE       #17 0x56120ce7ff34 <unknown>\nE       #18 0x7fcb15c94ac3 <unknown>\n\n/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/selenium/webdriver/support/wait.py:95: TimeoutException",
  "flaky" : false,
  "newFailed" : false,
  "beforeStages" : [ {
    "name" : "_session_faker",
    "time" : {
      "start" : 1707613764728,
      "stop" : 1707613764790,
      "duration" : 62
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "driver",
    "time" : {
      "start" : 1707614078488,
      "stop" : 1707614079442,
      "duration" : 954
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "clear_allure_result_folder",
    "time" : {
      "start" : 1707613764790,
      "stop" : 1707613764791,
      "duration" : 1
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  } ],
  "testStage" : {
    "description" : "Check that Contact Us link in footer is correct",
    "status" : "broken",
    "statusMessage" : "selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x56120ce80d93 <unknown>\n#1 0x56120cb64337 <unknown>\n#2 0x56120cbafd60 <unknown>\n#3 0x56120cbafe51 <unknown>\n#4 0x56120cbf3bc4 <unknown>\n#5 0x56120cbd246d <unknown>\n#6 0x56120cbf112d <unknown>\n#7 0x56120cbd21e3 <unknown>\n#8 0x56120cba2135 <unknown>\n#9 0x56120cba313e <unknown>\n#10 0x56120ce44e4b <unknown>\n#11 0x56120ce48dfa <unknown>\n#12 0x56120ce316d5 <unknown>\n#13 0x56120ce49a6f <unknown>\n#14 0x56120ce1569f <unknown>\n#15 0x56120ce6e098 <unknown>\n#16 0x56120ce6e262 <unknown>\n#17 0x56120ce7ff34 <unknown>\n#18 0x7fcb15c94ac3 <unknown>",
    "statusTrace" : "self = <test_footer.TestFooter object at 0x7f4e6057e350>\ndriver = <selenium.webdriver.chrome.webdriver.WebDriver (session=\"d5cfcff0714348071d8c97ebbeacd3f8\")>\nURL = 'https://magento.softwaretestingboard.com/what-is-new.html'\n\n    @allure.title(\"TC 02.02.05 - Check Contact Us link in footer leads to the correct page \"\n                  \"from each page specified in DATA_1\")\n    @pytest.mark.parametrize('URL', DATA_1)\n    def test_tc_02_02_05_check_contact_us_link_functionality(self, driver, URL):\n        \"\"\"Check that Contact Us link in footer is correct\"\"\"\n        page = FooterPage(driver, url=URL)\n        page.open()\n>       page.check_contact_us_link_functionality()\n\ntests/test_footer.py:557: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \npages/footer_page.py:326: in check_contact_us_link_functionality\n    contact_us_link = self.element_is_visible(self.footer_locators.CONTACT_US_LINK)\npages/base_page.py:29: in element_is_visible\n    self.go_to_element(self.element_is_present(locator))\npages/base_page.py:50: in element_is_present\n    return wait(self.driver, timeout).until(EC.presence_of_element_located(locator))\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"d5cfcff0714348071d8c97ebbeacd3f8\")>\nmethod = <function presence_of_element_located.<locals>._predicate at 0x7f4e60926700>\nmessage = ''\n\n    def until(self, method, message: str = \"\"):\n        \"\"\"Calls the method provided with the driver as an argument until the \\\n        return value does not evaluate to ``False``.\n    \n        :param method: callable(WebDriver)\n        :param message: optional message for :exc:`TimeoutException`\n        :returns: the result of the last call to `method`\n        :raises: :exc:`selenium.common.exceptions.TimeoutException` if timeout occurs\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.monotonic() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, \"screen\", None)\n                stacktrace = getattr(exc, \"stacktrace\", None)\n            time.sleep(self._poll)\n            if time.monotonic() > end_time:\n                break\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message: \nE       Stacktrace:\nE       #0 0x56120ce80d93 <unknown>\nE       #1 0x56120cb64337 <unknown>\nE       #2 0x56120cbafd60 <unknown>\nE       #3 0x56120cbafe51 <unknown>\nE       #4 0x56120cbf3bc4 <unknown>\nE       #5 0x56120cbd246d <unknown>\nE       #6 0x56120cbf112d <unknown>\nE       #7 0x56120cbd21e3 <unknown>\nE       #8 0x56120cba2135 <unknown>\nE       #9 0x56120cba313e <unknown>\nE       #10 0x56120ce44e4b <unknown>\nE       #11 0x56120ce48dfa <unknown>\nE       #12 0x56120ce316d5 <unknown>\nE       #13 0x56120ce49a6f <unknown>\nE       #14 0x56120ce1569f <unknown>\nE       #15 0x56120ce6e098 <unknown>\nE       #16 0x56120ce6e262 <unknown>\nE       #17 0x56120ce7ff34 <unknown>\nE       #18 0x7fcb15c94ac3 <unknown>\n\n/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/selenium/webdriver/support/wait.py:95: TimeoutException",
    "steps" : [ {
      "name" : "Open a browser",
      "time" : {
        "start" : 1707614079443,
        "stop" : 1707614080348,
        "duration" : 905
      },
      "status" : "passed",
      "steps" : [ ],
      "attachments" : [ ],
      "parameters" : [ ],
      "stepsCount" : 0,
      "attachmentsCount" : 0,
      "shouldDisplayMessage" : false,
      "hasContent" : false
    }, {
      "name" : "Check Contact Us link leads to the correct page",
      "time" : {
        "start" : 1707614080349,
        "stop" : 1707614085426,
        "duration" : 5077
      },
      "status" : "broken",
      "statusMessage" : "selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x56120ce80d93 <unknown>\n#1 0x56120cb64337 <unknown>\n#2 0x56120cbafd60 <unknown>\n#3 0x56120cbafe51 <unknown>\n#4 0x56120cbf3bc4 <unknown>\n#5 0x56120cbd246d <unknown>\n#6 0x56120cbf112d <unknown>\n#7 0x56120cbd21e3 <unknown>\n#8 0x56120cba2135 <unknown>\n#9 0x56120cba313e <unknown>\n#10 0x56120ce44e4b <unknown>\n#11 0x56120ce48dfa <unknown>\n#12 0x56120ce316d5 <unknown>\n#13 0x56120ce49a6f <unknown>\n#14 0x56120ce1569f <unknown>\n#15 0x56120ce6e098 <unknown>\n#16 0x56120ce6e262 <unknown>\n#17 0x56120ce7ff34 <unknown>\n#18 0x7fcb15c94ac3 <unknown>\n\n",
      "statusTrace" : "  File \"/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/home/runner/work/luma_project/luma_project/pages/footer_page.py\", line 326, in check_contact_us_link_functionality\n    contact_us_link = self.element_is_visible(self.footer_locators.CONTACT_US_LINK)\n                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/home/runner/work/luma_project/luma_project/pages/base_page.py\", line 29, in element_is_visible\n    self.go_to_element(self.element_is_present(locator))\n                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/home/runner/work/luma_project/luma_project/pages/base_page.py\", line 50, in element_is_present\n    return wait(self.driver, timeout).until(EC.presence_of_element_located(locator))\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/selenium/webdriver/support/wait.py\", line 95, in until\n    raise TimeoutException(message, screen, stacktrace)\n",
      "steps" : [ {
        "name" : "Find a visible element",
        "time" : {
          "start" : 1707614080349,
          "stop" : 1707614085426,
          "duration" : 5077
        },
        "status" : "broken",
        "statusMessage" : "selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x56120ce80d93 <unknown>\n#1 0x56120cb64337 <unknown>\n#2 0x56120cbafd60 <unknown>\n#3 0x56120cbafe51 <unknown>\n#4 0x56120cbf3bc4 <unknown>\n#5 0x56120cbd246d <unknown>\n#6 0x56120cbf112d <unknown>\n#7 0x56120cbd21e3 <unknown>\n#8 0x56120cba2135 <unknown>\n#9 0x56120cba313e <unknown>\n#10 0x56120ce44e4b <unknown>\n#11 0x56120ce48dfa <unknown>\n#12 0x56120ce316d5 <unknown>\n#13 0x56120ce49a6f <unknown>\n#14 0x56120ce1569f <unknown>\n#15 0x56120ce6e098 <unknown>\n#16 0x56120ce6e262 <unknown>\n#17 0x56120ce7ff34 <unknown>\n#18 0x7fcb15c94ac3 <unknown>\n\n",
        "statusTrace" : "  File \"/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/home/runner/work/luma_project/luma_project/pages/base_page.py\", line 29, in element_is_visible\n    self.go_to_element(self.element_is_present(locator))\n                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/home/runner/work/luma_project/luma_project/pages/base_page.py\", line 50, in element_is_present\n    return wait(self.driver, timeout).until(EC.presence_of_element_located(locator))\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/selenium/webdriver/support/wait.py\", line 95, in until\n    raise TimeoutException(message, screen, stacktrace)\n",
        "steps" : [ {
          "name" : "Find a present element",
          "time" : {
            "start" : 1707614080349,
            "stop" : 1707614085426,
            "duration" : 5077
          },
          "status" : "broken",
          "statusMessage" : "selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x56120ce80d93 <unknown>\n#1 0x56120cb64337 <unknown>\n#2 0x56120cbafd60 <unknown>\n#3 0x56120cbafe51 <unknown>\n#4 0x56120cbf3bc4 <unknown>\n#5 0x56120cbd246d <unknown>\n#6 0x56120cbf112d <unknown>\n#7 0x56120cbd21e3 <unknown>\n#8 0x56120cba2135 <unknown>\n#9 0x56120cba313e <unknown>\n#10 0x56120ce44e4b <unknown>\n#11 0x56120ce48dfa <unknown>\n#12 0x56120ce316d5 <unknown>\n#13 0x56120ce49a6f <unknown>\n#14 0x56120ce1569f <unknown>\n#15 0x56120ce6e098 <unknown>\n#16 0x56120ce6e262 <unknown>\n#17 0x56120ce7ff34 <unknown>\n#18 0x7fcb15c94ac3 <unknown>\n\n",
          "statusTrace" : "  File \"/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/home/runner/work/luma_project/luma_project/pages/base_page.py\", line 50, in element_is_present\n    return wait(self.driver, timeout).until(EC.presence_of_element_located(locator))\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/selenium/webdriver/support/wait.py\", line 95, in until\n    raise TimeoutException(message, screen, stacktrace)\n",
          "steps" : [ ],
          "attachments" : [ ],
          "parameters" : [ {
            "name" : "locator",
            "value" : "('css selector', '.footer div :nth-child(3) a')"
          }, {
            "name" : "timeout",
            "value" : "5"
          } ],
          "stepsCount" : 0,
          "attachmentsCount" : 0,
          "shouldDisplayMessage" : true,
          "hasContent" : true
        } ],
        "attachments" : [ ],
        "parameters" : [ {
          "name" : "locator",
          "value" : "('css selector', '.footer div :nth-child(3) a')"
        }, {
          "name" : "timeout",
          "value" : "10"
        } ],
        "stepsCount" : 1,
        "attachmentsCount" : 0,
        "shouldDisplayMessage" : false,
        "hasContent" : true
      } ],
      "attachments" : [ ],
      "parameters" : [ ],
      "stepsCount" : 2,
      "attachmentsCount" : 0,
      "shouldDisplayMessage" : false,
      "hasContent" : true
    } ],
    "attachments" : [ {
      "uid" : "93d5d1a3b0b658fe",
      "name" : "Screenshot",
      "source" : "93d5d1a3b0b658fe.png",
      "type" : "image/png",
      "size" : 545595
    }, {
      "uid" : "5ee1b7fe6ae15248",
      "name" : "HTML source",
      "source" : "5ee1b7fe6ae15248.html",
      "type" : "text/html",
      "size" : 115140
    } ],
    "parameters" : [ ],
    "stepsCount" : 4,
    "attachmentsCount" : 2,
    "shouldDisplayMessage" : true,
    "hasContent" : true
  },
  "afterStages" : [ {
    "name" : "driver::0",
    "time" : {
      "start" : 1707614085635,
      "stop" : 1707614085703,
      "duration" : 68
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "clear_allure_result_folder::0",
    "time" : {
      "start" : 1707615033978,
      "stop" : 1707615033978,
      "duration" : 0
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  } ],
  "labels" : [ {
    "name" : "epic",
    "value" : "Test Footer"
  }, {
    "name" : "parentSuite",
    "value" : "tests"
  }, {
    "name" : "suite",
    "value" : "test_footer"
  }, {
    "name" : "subSuite",
    "value" : "TestFooter"
  }, {
    "name" : "host",
    "value" : "fv-az1118-65"
  }, {
    "name" : "thread",
    "value" : "3041-MainThread"
  }, {
    "name" : "framework",
    "value" : "pytest"
  }, {
    "name" : "language",
    "value" : "cpython3"
  }, {
    "name" : "package",
    "value" : "tests.test_footer"
  }, {
    "name" : "resultFormat",
    "value" : "allure2"
  } ],
  "parameters" : [ {
    "name" : "URL",
    "value" : "'https://magento.softwaretestingboard.com/what-is-new.html'"
  } ],
  "links" : [ ],
  "hidden" : false,
  "retry" : false,
  "extra" : {
    "severity" : "normal",
    "retries" : [ ],
    "categories" : [ {
      "name" : "Test defects",
      "matchedStatuses" : [ ],
      "flaky" : false
    } ],
    "tags" : [ ]
  },
  "source" : "b2fd2386b1da2c23.json",
  "parameterValues" : [ "'https://magento.softwaretestingboard.com/what-is-new.html'" ]
}